packages/insomnia-app/app/models/stats.js

```diff
@@ -49,3 +56,40 @@ export async function get(): Promise<Stats> {
     return results[0];
   }
 }
+
+export async function incrementRequestStats({
+  createdRequests,
+  deletedRequests,
+  executedRequests,
+}: $Shape<Stats>) {
+  const stats = await get();
+  await update({
+    ...(createdRequests && { createdRequests: stats.createdRequests + createdRequests }),
+    ...(deletedRequests && { deletedRequests: stats.deletedRequests + deletedRequests }),
+    ...(executedRequests && { executedRequests: stats.executedRequests + executedRequests }),
+  });
+}
+
+export async function incrementCreatedRequests() {
+  await incrementRequestStats({ createdRequests: 1 });
+}
+
+export async function incrementDeletedRequests() {
+  await incrementRequestStats({ deletedRequests: 1 });
+}
+
+export async function incrementExecutedRequests() {
+  await incrementRequestStats({ executedRequests: 1 });
+}
+
+export async function incrementCreatedRequestsForDescendents(doc: BaseModel) {
```

Comment:
I would favor restricting the types here because we know for sure this should only be a request group or a workspace (same with the deletion function below)

```suggestion
export async function incrementCreatedRequestsForDescendents(doc: Workspace | RequestGroup) {
```

Reply:
Sounds reasonable 👍